src/Framework/Process.cxx

```diff
@@ -105,9 +105,17 @@ Process::Process(const framework::config::Parameters &configuration)
     conditions_.createConditionsObjectProvider(className, objectName, tagName,
                                                cop);
   }
+
+  bool logPerformance = configuration.getParameter<bool>("logPerformance", false);
+  if (logPerformance) {
+    performance_ = new performance::Tracker(makeHistoDirectory("performance"));
+  }
 }
 
 Process::~Process() {
+  // need to delete the performance object so that it is
+  // written before we close the histogram file below
+  if (performance_) delete performance_;
```

Comment:
I tried using a unique_ptr and the issue arises from the `std::move` in `getDummy`. We'd need to define a move constructor to get it working and, since I need to call `reset` anyways to insure that it is destructed at the right time, I opted for a raw pointer.